helpers/load_test_interface.go

```diff
@@ -168,6 +169,90 @@ func WRK2LoadTest(opts *models.LoadTestOptions) (map[string]interface{}, *period
 	return resultsMap, result, nil
 }
 
+// NighthawkLoadTest is the actual code which invokes nighthawk to run the load test
+func NighthawkLoadTest(opts *models.LoadTestOptions) (map[string]interface{}, *periodic.RunnerResults, error) {
+	qps := opts.HTTPQPS
+
+	if qps <= 0 {
+		qps = -1 // 0==unitialized struct == default duration, -1 (0 for flag) is max
+	}
+
+	rURL := strings.TrimLeft(opts.URL, " \t\r\n")
+
+	ro := &apinighthawk.NighthawkConfig{
+		DurationInSeconds: opts.Duration.Seconds(),
+		Thread:            opts.HTTPNumThreads,
+		QPS:        qps,
+		URL:         rURL,
+	}
+
+	// var res periodic.HasRunnerResult
+	var err error
+
+	if ( opts.SupportedLoadTestMethods == 2 ) {
+		err := errors.New("Nighthawk does not support gRPC load testing")
+		logrus.Error(err)
+		return nil, nil, err
+	}
+
+	res, err := apinighthawk.NighthawkRun(ro)
+	if err != nil {
+		err = errors.Wrap(err, "error while running tests")
+		logrus.Error(err)
+		return nil, nil, err
+	}
+
+	// result := string(res)
+
+
+	logrus.Debugf("original version of the test: %+#v", res)
+
+	var result *periodic.RunnerResults
+	// var bd []byte
+	// if ( opts.SupportedLoadTestMethods == 2 ) {
+	// 	gres, _ := res.(*fgrpc.GRPCRunnerResults)
+	// 	bd, err = json.Marshal(gres)
+	// 	result = gres.Result()
+	// } else {
+	// 	hres, _ := res.(*fhttp.HTTPRunnerResults)
+	// 	bd, err = json.Marshal(hres)
+	// 	result = hres.Result()
+	// }
+	// if err != nil {
+	// 	err = errors.Wrap(err, "error while converting results to map")
+	// 	logrus.Error(err)
+	// 	return nil, nil, err
+	// }
+
+	bd, err := json.Marshal(res)
+
+	if err != nil {
+		err = errors.Wrap(err, "error while converting results to map")
+		logrus.Error(err)
+		return nil, nil, err
+	}
+
+	err = json.Unmarshal(res, result)
+
+	if err != nil {
+		err = errors.Wrap(err, "error while unmarshaling data to the FortioHTTPRunner")
+		logrus.Error(err)
+		return nil, nil, err
+	}
+
+	resultsMap := map[string]interface{}{}
+	err = json.Unmarshal(bd, &resultsMap)
+
+	if err != nil {
+		err = errors.Wrap(err, "error while unmarshaling data to map")
```

Comment:
Are these errors displayed as toaster notifications? If so, please cleanup.

Reply:
These are logs and displayed on stdout, this PR is for server side changes.